pkg/system/reconciler.go

```diff
@@ -435,12 +435,22 @@ func (r *Reconciler) Reconcile() (reconcile.Result, error) {
 			log.Warnf("⏳ Temporary Error: %s", err)
 		}
 	} else {
-		r.SetPhase(
-			nbv1.SystemPhaseReady,
-			"SystemPhaseReady",
-			"noobaa operator completed reconcile - system is ready",
-		)
-		log.Infof("✅ Done")
+		// Postgres upgrade failure workaround
+		// if the system reconciliation has no other error, but the postgres image is still postresql-12 image, set the status to Rejected
+		psql12Image, ok := os.LookupEnv("NOOBAA_PSQL_12_IMAGE")
+		if ok && r.NooBaaPostgresDB.Spec.Template.Spec.Containers[0].Image == psql12Image {
+			r.SetPhase(nbv1.SystemPhaseRejected,
+				"PostgresImageVersion",
```

Comment:
just maybe thinking of another name that is more indicative of an error? 
WrongPostgresImageVersion of OldPostgresImageVersion